docs/contribute/cumulative-docs/content-patterns.md

```diff
@@ -0,0 +1,383 @@
+* Try not to include information surrounding a code block in the tabs.
+  Make the tab content as small as possible apart from the procedure itself.
+
+**Example**:
+
+::::{image} ./images/example-code-block-tabs.png
+:screenshot:
+:alt:
+::::
+
+## Workflows vary [workflow]
+
+When one or more steps in a process differs.
+
+### Solution A: Use inline `applies_to` [workflow-inline]
+
+Using inline `applies_to` badges to a few line items in an ordered list is the lightest-touch solution,
+but might not be sufficient in all cases.
+
+**When to use inline `applies_to`**:
+
+* Workflow steps that vary between situations can be easily isolated.
+* Each step that varies, only varies between 3 or fewer situations (deployment types, versions, etc).
+* There are no more than 3 steps that need to be split into multiple lines with `applies_to` badges.
+
+**Best practices**:
+
+* Follow the [best practices for ordered and unordered lists](/contribute/cumulative-docs/best-practices.md#ordered-and-unordered-lists)
+  including the order of items and the placement of labels.
+
+**Example**: Only one item in an ordered list varies between Serverless and stateful.
+
+::::{image} ./images/workflow-inline.png
+:screenshot:
+:alt:
+::::
+
+### Solution B: Use tabs [workflow-tabs]
+
+Tabs are minimally disruptive in many situations.
+
+**When to use tabs**:
+
+* Using [inline `applies_to` badges](#workflow-inline) isn't appropriate.
+* All the tabs fit horizontally on a single row on a typical laptop screen.
+  This is usually around a maximum of four tabs.
+* The tab with the most content fits vertically on a typical laptop screen.
+  This is usually around of 20 lines.
+
+**Best practices**:
+
+* Try to minimize the number of tabs where possible. Try to work around small differences by
+  rewording or adding context in prose or in `note` style admonitions.
+* Try not to include information surrounding a procedure in the tabs.
+  Make the tab content as small as possible apart from the procedure itself.
+* Consider breaking up procedures into sets of procedures if only one section differs between contexts.
+
+**Example**:
+
+<image>
+
+### Solution C: Use sibling pages [workflow-sibling-pages]
+
+Sibling pages are a last resort when no other solutions are appropriate.
+
+**When to use sibling pages**:
+
+* Neither [inline `applies_to` badges](#workflow-inline) or [tabs](#workflow-tabs) are appropriate.
+* The workflow has significant differences across multiple procedures.
+* There are chained procedures where not all of the procedures are needed for all contexts
+  or where the flow across procedures is muddied when versioning context is added.
+* The workflow exists in a very complex page that is already heavily using tabs and other tools we use for versioning differences.
+  This makes it difficult to add another “layer” of content.
+% * Beta to GA transitions? Hide the beta doc and leave it linked to the GA doc, which will presumably be more stable?
```

Comment:
I like this idea. Can hide previous lifecycle state docs as a new lifecycle comes out (preview --> beta --> ga)